tests/serving/test_serving.py

```diff
@@ -644,6 +645,40 @@ def test_function():
     assert len(dummy_stream.event_list) == 1, "expected stream to get one message"
 
 
+def test_sampling_percentage():
+    fn = mlrun.new_function("tests", kind="serving")
+    fn.set_topology("router")
+    fn.add_model("my", ".", class_name=ModelTestingClass(multiplier=100))
+    random.seed(0)
+    random_sample_percentage = 50
+
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError) as err:
+        fn.set_tracking(stream_path="dummy://", sampling_percentage=101)
+        assert (
+            str(err.value)
+            == "`sampling_percentage` must be greater than 0 and less or equal to 100."
+        )
+
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError) as err:
+        fn.set_tracking(stream_path="dummy://", sampling_percentage=0)
+        assert (
+            str(err.value)
+            == "`sampling_percentage` must be greater than 0 and less or equal to 100."
+        )
+
+    fn.set_tracking(
+        stream_path="dummy://", sampling_percentage=random_sample_percentage
+    )
+    server = fn.to_mock_server()
+    for i in range(500):
+        server.test("/v2/models/my/infer", testdata)
```

Comment:
testdata = '{"inputs": [5]}'

You can change it to '{"inputs": [5, 7]}' and to test the single infer is working as expected.

Reply:
I'll add more invocations with multiple features and verify that the estimated count is correct